src/basic/cgroup-util.c

```diff
@@ -1198,6 +1198,38 @@ int cg_path_get_unit(const char *path, char **ret) {
         return 0;
 }
 
+int cg_path_get_unit_path(const char *path, char **ret) {
+        char *path_copy, *unit_name, *unit_path;
+        size_t unit_name_len;
+
+        assert(path);
+        assert(ret);
+
+        path_copy = strdupa_safe(path);
+
+        unit_name = (char *)skip_slices(path_copy);
+        unit_name_len = strcspn(unit_name, "/");
+        if (unit_name_len < 3)
+                return -ENXIO;
+
+        unit_name[unit_name_len] = '\0';
+
+        /* If unit_name is escaped, overwrite it in place with the unespaced
+         * version. memmove() doesn't mind overlapping memory areas. */
+        memmove(unit_name, cg_unescape(unit_name), unit_name_len);
```

Comment:
why is the memmove needed here? just pass cg_unescape to unit_name_is_valid?

Reply:
We store a copy of `path` into `path_copy` so we can modify it: 
- Add `\0` after the unit's name.
- Remove escape character, if any.

Then, the modified `path_copy` is strdup()'ed into the return value. `memmove()` is used to ensure the escaping character is removed from `path_copy`.